code-studio/src/js/components/BeeCellEditor.jsx

```diff
@@ -0,0 +1,118 @@
+/**
+ * @overview React component to allow for easy editing and creation of
+ * BeeCells
+ * @see @cdo/apps/maze/beeCell
+ */
+/* global React */
+
+var BeeCell = require('@cdo/apps/maze/beeCell');
+
+var BeeCellEditor = module.exports = React.createClass({
+  propTypes: {
+    cell: React.PropTypes.object.isRequired,
+    row: React.PropTypes.number.isRequired,
+    col: React.PropTypes.number.isRequired,
+    onUpdate: React.PropTypes.func.isRequired,
+  },
+
+  handleChange: function (event) {
+    var values = {};
+    var nodes = this.getDOMNode().querySelectorAll('[name]');
+    for (var i = 0, node; (node = nodes[i]); i++) {
```

Comment:
hah, yeah, nodelists are a pain. One (very minor) disadvantage of both of those approaches is that in many modern broswers, `NodeList.length` is an O(n) operation, hence the rather silly nature of this for loop.